src/library/browse/browsethread.cpp

```diff
@@ -154,15 +153,16 @@ void BrowseThread::populateModel() {
 
         const QString filepath = fileIt.next();
         {
-            TrackPointer pTrack =
-                GlobalTrackCache::instance().resolve(
-                        filepath, thisPath.token()).getTrack();
-            // The GlobalTrackCache is unlocked instantly even if a new track object
-            // has been created and inserted into the cache. Newly created track
-            // objects will only contain a reference of the corresponding file,
-            // but not any metadata, yet. This reduces lock contention on the
-            // global track cache.
-
+            // We need to create a temporary track object and cannot
+            // use the GlobalTrackCache, because cached tracks must
+            // live in the main thread. There is a tiny chance that
+            // the file is removed (after exporting metadata) just
+            // in the moment when reading metadata from it.
+            // TODO: Resolve track object through GlobalTrackCache
+            // after multi-threaded library management is in place
+            // to ensure that only a single track object exists for
+            // each file.
+            TrackPointer pTrack = Track::newTemporary(filepath, thisPath.token());
             // Update the track object by (re-)importing metadata from the file
             SoundSourceProxy(pTrack).updateTrackFromSource();
```

Comment:
Here we allow reading a file while we may also just write the same file. I think that can be a source of a taglib crash.

Reply:
Very, very, very unlikely and I don't think this will happen. We are writing into a temporary file and try to delete the original file before moving the temporary file over it. The deletion of the original file should fail while it is read by TagLib.

Reply:
In general, there is the question, if we need to prevent concurrent reading and writing of the same file. The main reason was originally that the file was shifted under a reader while writing. Can this happen here? Should we move the initial meta read into the cache locking region?

Reply:
No, we are not modifying the original file, so the reader is unaffected. Only writing might fail when trying to delete the original file before moving the newly written file.

We are not able to use GlobalTrackCache here without risking to access the database in TrackDAO from the wrong thread. This is not possible until TrackDAO is able to dynamically obtain thread-local database connections from DbConnectionPool.

Reply:
> Only writing might fail when trying to delete the original file before moving the newly written file.

That would be odd as well, because we loose user data. The temporary file is not stored into the database, so the data base access this should be no issue.